docs/resources/samples/get-table-data.md

```diff
@@ -0,0 +1,259 @@
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook): TableData[] {
+  const table = workbook.getWorksheet('PlainTable').getTables()[0];
+  // If you know the table name, you can also do the following:
+  // const table = workbook.getTable('Table13436');
+  const texts = table.getRange().getTexts();
+  let returnObjects: TableData[] = [];
+  if (table.getRowCount() > 0)  {
+    returnObjects = returnObjectFromValues(texts);
+  } 
+  console.log(JSON.stringify(returnObjects));  
+  return returnObjects
+}
+
+function returnObjectFromValues(values: string[][]): TableData[] {
+  let objArray = [];
+  let objKeys: string[] = [];
+  for (let i = 0; i < values.length; i++) {
+    if (i === 0) {
+      objKeys = values[i]
+      continue;
+    }
+    let obj = {}
+    for (let j = 0; j < values[i].length; j++) {
+      obj[objKeys[j]] = values[i][j]
+    }
+    objArray.push(obj);
+  }
+  return objArray as TableData[];
+}
+
+interface BasicObj {
+  [key: string]: string
+}
+
+interface TableData {
+  "Event ID": string
+  Date: string
+  Location: string
+  Capacity: string
+  Speakers: string
+}
+```
+
+#### Sample output
+
+```json
+[{
+    "Event ID": "E107",
+    "Date": "2020-12-10",
+    "Location": "Montgomery",
+    "Capacity": "10",
+    "Speakers": "Debra Berger"
+}, {
+    "Event ID": "E108",
+    "Date": "2020-12-11",
+    "Location": "Montgomery",
+    "Capacity": "10",
+    "Speakers": "Delia Dennis"
+}, {
+    "Event ID": "E109",
+    "Date": "2020-12-12",
+    "Location": "Montgomery",
+    "Capacity": "10",
+    "Speakers": "Diego Siciliani"
+}, {
+    "Event ID": "E110",
+    "Date": "2020-12-13",
+    "Location": "Boise",
+    "Capacity": "25",
+    "Speakers": "Gerhart Moller"
+}, {
+    "Event ID": "E111",
+    "Date": "2020-12-14",
+    "Location": "Salt Lake City",
+    "Capacity": "20",
+    "Speakers": "Grady Archie"
+}, {
+    "Event ID": "E112",
+    "Date": "2020-12-15",
+    "Location": "Fremont",
+    "Capacity": "25",
+    "Speakers": "Irvin Sayers"
+}, {
+    "Event ID": "E113",
+    "Date": "2020-12-16",
+    "Location": "Salt Lake City",
+    "Capacity": "20",
+    "Speakers": "Isaiah Langer"
+}, {
+    "Event ID": "E114",
+    "Date": "2020-12-17",
+    "Location": "Salt Lake City",
+    "Capacity": "20",
+    "Speakers": "Johanna Lorenz"
+}]
+```
+
+### Return table data as JSON with hyperlink text
+
+#### Screenshots
+
+![Screenshot displaying table data that includes hyperlinks](../../images/table-hyperlink-view.png)
+![Screenshot displaying dialog to edit hyperlink](../../images/table-hyperlink-edit.png)
+
+#### Script
+
+> [!NOTE]
+> The script always extracts hyperlinks from the 4th column (0 index) of the table. You can change that order or include multiple columns as hyperlink data by modifying the code under the comment `// For the 4th column (0 index), extract the hyperlink and use that instead of text.`
+
+```TypeScript
```

Comment:
add code comments

Reply:
Yeah, I had received this feedback before. We're tracking with this item: 4958338